### PR TITLE
fix: Add /docs to Link hrefs on markdown script

### PR DESCRIPTION
**Pull request by jeremenichelli**
In this PR:
 - Apply another function to links coming from Next's `Link` component to add the base path of the application on Markdown scripts.

## Summary by CodeRabbit

* **Improvements**
  * Documentation links in guides now properly apply base path prefixes consistently, ensuring correct routing across different deployment environments

* **Tests**
  * Added comprehensive test coverage for link path handling, validating behavior with root-relative paths, external URLs, anchors, relative paths, and various edge cases

### CODE REVIEW

File: apps/docs/internals/internal-links.ts [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+ * `/guides/foo` lose the `/docs` prefix the deployed site adds.
+ *
+ * Returns the href unchanged when it's already prefixed, external, anchor-
+ * only, or relative (`./`, `../`).
+ */
+export function withDocsBasePath(href: string): string {
+  if (!href.startsWith('/')) return href
+  if (href.startsWith('//')) return href
+  if (href === DOCS_BASE_PATH || href.startsWith(`${DOCS_BASE_PATH}/`)) return href
+  return `${DOCS_BASE_PATH}${href}`
```
[the remark this answers is not in the document]
Comment (ChrisChinchilla) [reply apps/docs/internals/internal-links.ts:16]: @jeremenichelli is this valid?

[the remark this answers is not in the document]
Comment (jeremenichelli) [reply apps/docs/internals/internal-links.ts:16]: Hmmm yes and no. The liability is there but their examples and solutions don't. I will explore this better.

[the remark this answers is not in the document]
Comment (jeremenichelli) [reply apps/docs/internals/internal-links.ts:16]: @ChrisChinchilla found only one case, in `vector-columns.mdx` file:

```md
[supported language library](/docs#client-libraries)
```

The result is exactly the same, so no idea what this tool is saying. Also, this is only used in Link components which won't be use for internal sections or query modifications. So it's safe to say this comment is not valid.

**Review by ChrisChinchilla — APPROVED** [review]
